classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -0,0 +1,303 @@
+	 * Check if a navigationMenu exists with the given title.
+	 * @param $title int
+	 * @param $contextId int
+	 * @return boolean
+	 */
+	function navigationMenuExistsByTitle($title, $contextId) {
+		$result = $this->retrieve(
+			'SELECT COUNT(*)
+			FROM	navigation_menus
+			WHERE	title = ? AND
+				context_id = ?',
+			array(
+				$title,
+				(int) $contextId
+			)
+		);
+		$returner = isset($result->fields[0]) && $result->fields[0] != 0 ? true : false;
+
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Get the locale field names.
+	 * @return array
+	 */
+	function getLocaleFieldNames() {
+	    return array();
+	}
+
+	/**
+	 * Internal function to return an NavigationMenu object from a row.
+	 * @param $row array
+	 * @return NavigationMenu
+	 */
+	function _fromRow($row) {
+		$navigationMenu = $this->newDataObject();
+		$navigationMenu->setId($row['navigation_menu_id']);
+		$navigationMenu->setTitle($row['title']);
+		$navigationMenu->setAreaName($row['area_name']);
+		$navigationMenu->setContextId($row['context_id']);
+		$navigationMenu->setDefault($row['is_default']);
+
+		return $navigationMenu;
+	}
+
+	/**
+	 * Insert a new NavigationMenu.
+	 * @param $navigationMenu NavigationMenu
+	 * @return int
+	 */
+	function insertObject($navigationMenu) {
+		$this->update(
+				'INSERT INTO navigation_menus
+				(title, area_name, context_id, is_default)
+				VALUES
+				(?, ?, ?, ?)',
+			array(
+				$navigationMenu->getTitle(),
+				$navigationMenu->getAreaName(),
+				(int) $navigationMenu->getContextId(),
+				(int) $navigationMenu->getDefault(),
+			)
+		);
+		$navigationMenu->setId($this->getInsertId());
+
+		return $navigationMenu->getId();
+	}
+
+	/**
+	 * Update an existing NavigationMenu
+	 * @param NavigationMenu $navigationMenu
+	 * @return boolean
+	 */
+	function updateObject($navigationMenu) {
+		$returner = $this->update(
+			'UPDATE	navigation_menus
+			SET	title = ?,
+				area_name = ?,
+				context_id = ?,
+				is_default = ?
+			WHERE	navigation_menu_id = ?',
+			array(
+				$navigationMenu->getTitle(),
+				$navigationMenu->getAreaName(),
+				(int) $navigationMenu->getContextId(),
+				(int) $navigationMenu->getDefault(),
+				(int) $navigationMenu->getId(),
+			)
+		);
+
+		return $returner;
+	}
+
+	/**
+	 * Delete a NavigationMenu.
+	 * TODO::defstat - What whould we do with NavigationMenuItems having the deleted NavigationMenu as parent
+	 * @param $navigationMenu NavigationMenu
+	 * @return boolean
+	 */
+	function deleteObject($navigationMenu) {
+		return $this->deleteById($navigationMenu->getId());
+	}
+
+	/**
+	 * Delete a NavigationMenu.
+	 * @param $navigationMenuId int
+	 */
+	function deleteById($navigationMenuId) {
+		$this->update('DELETE FROM navigation_menus WHERE navigation_menu_id = ?', (int) $navigationMenuId);
+
+		$navigationMenuItemAssignmentDao = DAORegistry::getDAO('NavigationMenuItemAssignmentDAO');
+		$navigationMenuItemAssignmentDao->deleteByMenuId($navigationMenuId);
+
+		return true;
```

Comment:
This return value is undocumented and doesn't add any value. Suggest removing it.